great_expectations/cli/suite.py

```diff
@@ -282,26 +401,14 @@ def suite_edit(
     context: DataContext = ctx.obj.data_context
     usage_event_end: str = ctx.obj.usage_event_end
 
-    error_message: Optional[str] = None
-
-    if not interactive and (
-        (datasource_name is not None) or (batch_request is not None)
-    ):
-        error_message = """Using --datasource-name DATASOURCE_NAME option or --batch-request <path to JSON file> \
-option requires --interactive flag.
-"""
-
-    if interactive and (datasource_name is not None) and (batch_request is not None):
-        error_message = """Only one of --datasource-name DATASOURCE_NAME and --batch-request <path to JSON file> \
-options can be used.
-"""
-
-    if error_message is not None:
-        cli_message(string=f"<red>{error_message}</red>")
-        toolkit.send_usage_message(
-            data_context=context, event=usage_event_end, success=False
-        )
-        sys.exit(1)
+    interactive = _process_suite_edit_flags_and_prompt(
```

Comment:
Same comment as about -- about moving this process flags method to a toolkit type module...